compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -1960,12 +1965,11 @@ public void visit(BLangSimpleVarRef bLangSimpleVarRef) {
                 bLangSimpleVarRef.symbol = symbol;
                 bLangSimpleVarRef.varSymbol = symbol;
             }
-        } else if (resolvedSymbol != symTable.notFoundSymbol && symbol != null) {
+        } else if (resolvedSymbol != symTable.notFoundSymbol) {
```

Comment:
Before performing this check, we should confirm whether the resolved symbol is a closure. 
In the following example, the closure flag of the resolved symbol is set to true when comes to this logic. Therefore, we need to do a check to determine whether it is a closure variable or not.
```ballerina
import ballerina/io;

public function main() {
    foo();
}

function foo() {
    int? i = 3;
    function k = function () {
        int? j = i;
    };
    _ = i is () ? 2 : from var _ in [1, 2]
                        where i + 2 == 3
                        select 2;
    io:println(i);
}
```

Reply:
Please add a test for this as well.